async_fuse/src/memfs/fs_util.rs

```diff
@@ -271,23 +273,32 @@ pub fn convert_to_fuse_attr(attr: FileAttr) -> FuseAttr {
 }
 
 /// Helper funtion to load directory data
-pub async fn load_dir_data(fd: RawFd) -> anyhow::Result<BTreeMap<OsString, DirEntry>> {
-    let dir = blocking!(Dir::from_fd(fd)).context(format!("failed to build Dir from fd={}", fd))?;
-    let dir_entry_map = blocking!(
-        let dir_entry_map: BTreeMap<OsString, DirEntry> = dir
-            .filter_map(std::result::Result::ok) // filter out error result
-            .filter(|e| {
-                let bytes = e.entry_name().as_bytes();
-                !bytes.starts_with(&[b'.']) // skip hidden entries, '.' and '..'
-            })
-            .filter_map(|e| match e.entry_type() {
-                SFlag::S_IFDIR | SFlag::S_IFREG | SFlag::S_IFLNK => Some((e.entry_name().into(), e)),
-                _ => None,
-            })
-            .collect();
-        dir_entry_map
-    );
-    Ok(dir_entry_map)
+pub async fn load_dir_data(dirfd: RawFd) -> anyhow::Result<BTreeMap<OsString, DirEntry>> {
+    unblock(move || {
+        let dir = {
+            // new_fd = openat(dirfd, ".", ...)
+            // dirp = fdopendir(new_fd)
+            // it is safe because the new fd is not used by anything other than dirp
```

Comment:
Remove it.

Reply:
pingcap style guide explains why here needs some comments.

<https://pingcap.github.io/style-guide/rust/unsafe.html>

> Document every use of unsafe.
> Even trivial uses should have a comment stating why it is safe. Including unsafe impl.
> Document why unsafe blocks are safe and how Rust’s invariants are ensured.
> For unsafe functions, document the invariants the caller must ensure, and any restrictions on the input, or how the output may be used.